src/core/Tree/Tree.tsx

```diff
@@ -0,0 +1,228 @@
+    selectedNodes={selectedNodes}
+    {...args}
+  />
+ */
+export const Tree = (props: TreeProps) => {
+  const {
+    data,
+    className,
+    nodeRenderer,
+    expandedNodes,
+    selectedNodes,
+    ...rest
+  } = props;
+  useTheme();
+
+  const treeRef = React.useRef<HTMLUListElement>(null);
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLUListElement>) => {
+    const nodes: Array<HTMLElement> = Array.from(
+      treeRef.current?.querySelectorAll('.iui-tree-node:not(.iui-disabled)') ||
+        [],
+    );
+
+    if (!nodes.length) {
+      return;
+    }
+
+    let newIndex = -1;
+    const currentIndex = nodes.findIndex(
+      (node) => node === treeRef.current?.ownerDocument.activeElement,
+    );
+
+    const expander = nodes[currentIndex].querySelector(
+      '.iui-tree-node-content-expander-icon',
+    ) as HTMLElement;
+
+    switch (event.key) {
+      case 'ArrowDown': {
+        newIndex = currentIndex + 1;
+        break;
+      }
+      case 'ArrowUp': {
+        newIndex = currentIndex - 1;
+        break;
+      }
+      case 'ArrowLeft':
+        if (
+          expander === null ||
+          !expander?.classList.contains(
+            'iui-tree-node-content-expander-icon-expanded',
+          )
+        ) {
+          newIndex = currentIndex - 1;
+        } else {
+          expander.parentElement?.click();
+        }
+        break;
+      case 'ArrowRight':
+        if (
+          expander === null ||
+          expander?.classList.contains(
+            'iui-tree-node-content-expander-icon-expanded',
+          )
+        ) {
+          newIndex = currentIndex + 1;
+        } else {
+          expander.parentElement?.click();
+          return;
+        }
+        break;
+      case 'Enter':
+      case ' ':
+      case 'Spacebar':
+        nodes[currentIndex].click();
+        event.preventDefault();
+        return;
+    }
+
+    if (newIndex >= 0 && newIndex < nodes.length) {
+      nodes[newIndex].focus();
+    }
+  };
+
+  const [selected, setSelected] = React.useState<Array<string>>([]);
+  const [expanded, setExpanded] = React.useState<Array<string>>([]);
+
+  const flatNodesList = React.useMemo(() => {
+    const flatList: FlatNode[] = [];
+    const flatNodes = (
+      nodes: TreeData[] = [],
+      parent?: FlatNode,
+      depth = 0,
+    ) => {
+      const parentId = parent?.id ?? null;
+      const nodeIdList = Array<string>();
+      nodes.forEach((node: TreeData, index) => {
+        const id = parentId ? `${parentId}-${index}` : `${index}`;
+        const nodeWrapper: FlatNode = {
+          id,
+          data: node,
+          depth,
+          subNodeIds: [],
+          parent: parent ?? undefined,
+        };
+        flatList.push(nodeWrapper);
+        nodeIdList.push(id);
+        if (node.subnodes && node.subnodes.length) {
+          const subNodeIds = flatNodes(node.subnodes, nodeWrapper, depth + 1);
+          nodeWrapper.subNodeIds = subNodeIds;
```

Comment:
Not sure anymore whether we should generate `subNodeIds` here.

Reply:
Oh, I see you are generating them inside `TreeNode`, so yeah, you can remove it from here.

Reply:
Removed